core/src/com/bot4s/telegram/future/Polling.scala

```diff
@@ -2,22 +2,25 @@ package com.bot4s.telegram.future
 
 import com.bot4s.telegram.api.{ Polling => BasePolling }
 import com.bot4s.telegram.methods.{ DeleteWebhook, GetMe }
-import com.bot4s.telegram.models.{ Update, User }
+import com.bot4s.telegram.models.User
 import com.typesafe.scalalogging.StrictLogging
 
 import scala.concurrent.Future
 import scala.util.control.NonFatal
+import com.bot4s.telegram.models.ParsedUpdate
 
 trait Polling extends BasePolling[Future] with BotExecutionContext with StrictLogging {
 
-  private type OffsetUpdates = (Option[Long], Seq[Update], User)
+  private type OffsetUpdates = (Option[Long], Seq[ParsedUpdate], User)
```

Comment:
We need to store the failure as well here because this is how we retrieve the max offset